app/views/rules/_handler.html.erb

```diff
@@ -23,7 +23,7 @@
     <div class="Polaris-FormLayout">
       <%= f.polaris_select(:service_name, available_handlers.map { |handler| [handler.label, handler.to_s] }, { prompt: "Select an action" }, { data: { handler: true } }) %>
 
-      <%= f.fields_for :settings, f.object.settings do |ff| %>
```

Comment:
Other than guessing, I couldn't explain with a reproductible fact why this used to work in the first place and what made rails 7.0 upgrade break it. 

It shouldn't have worked before since `f.object.settings` is a `Hash` and should have been treated as options, not as a record object....